# Genome assembly and analysis of Durio zibethinus
[Genome Analysis (1MB462)](https://www.uu.se/en/admissions/freestanding-courses/course-syllabus/?kKod=1MB462&lasar=), [Uppsala University](https://www.uu.se/en) - Author: Daniel Ågstrand

##
During this project a genome de novo assembly of the Musang King cultivars of Durian, Durio zibethinus species will be performed using PacBio reads which will be corrected with shorter Illumina reads. The quality of the reads will first be assessed and low quality reads will be removed with the help of Canu from the PAcBio reads and the quality of the Illumina reads will be checked with FastQC. If necessary the reads will also go through adapters trimming where the adapter sequence from the 3’-ends will be removed. When this is done the good quality PacBio long reads will be used with Canu to assemble the genome and then the quality of the assembly will be assessed. The trimmed Illumina short read will be assembled thoughter with the assembled PacBio genome as reference with Pilon. Afterward RNA-seq Illumuna short reads from Musang King, Durio zibethinus will be assembled to perform a transcriptome assembly with Trinity1. Then the assembled genome and transcriptome will then be used to perform an automatic structural and functional annotation with the help from Maker2 which also will be manual curated to determine the functional regions and identify genes responsible for producing sulphur odor molecules. The gene products that will be identified as the sulphur odor molecules will later be blast against gene products of the Monthong cultivars of the same species and against Allium cepa (onions), which is also known to produce sulphur molecules to hopefully be able to verify the annotation. A differential expression analysis will then be performed to identify how different regions are expressed under different conditions. First the RNA-Seq reads needs to be mapped to the assembled genome with the software Tophat, then the data from this will be used in HTSeq to count the reads from the mapped geneome. Last the software DESeq2 will be used to calculates the variance-mean dependence in the counted reads expressed under different conditions.
